### ISSUE TITLE
Support for failsafe?

### ISSUE THREAD

**Issue by orpiske** [mvndaemon/mvnd#86]
I noticed that our integration tests do not run when using mvnd. 

Not knowing anything Maven internals and the challenges of running these in parallel, I am not sure if: 
- it's a bug, 
- working as designed 
- a problem on our own build configuration. 

Could you, please, share some insights? Should the integration test execution, managed by the failsafe plugin, be supported?

**Comment by ppalaga** [comment]
Why do you think they do not run? There is https://github.com/mvndaemon/mvnd/issues/21 but tests not running at all would be new to me.

**Comment by orpiske** [comment]
Looking at our build output, there's a few strange things:

1. I have several reports of 0 tests run:
`[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0`


2. The log message says the tests were skipped
```
[INFO] --- maven-failsafe-plugin:2.22.2:integration-test (integration-test) @ itests-syslog ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-failsafe-plugin:2.22.2:verify (integration-test) @ itests-syslog ---
[INFO] Tests are skipped.
```

**Comment by ppalaga** [comment]
An integration tests reproducing the issue would be awesome.

**Comment by orpiske** [comment]
> An integration test reproducing the issue would be awesome.

I'll prepare a reproducer this week.

**Comment by famod** [comment]
@orpiske

>     1. [INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0

Is this from surefire? Are you using Junit5 parameterized tests?
If yes this _could_ be caused by: https://issues.apache.org/jira/browse/SUREFIRE-1741

**Comment by orpiske** [comment]
@famod this is for failsafe. We are not using parameterized tests. 

I am going to provide the reproducer soon. Today I have some time to focus on this item.

**Comment by orpiske** [comment]
Steps to reproduce: 
```
git clone https://github.com/orpiske/camel-kafka-connector.git --branch mvnd-issue-86
cd camel-kafka-connector
mvnd -U -DskipIntegrationTests=false clean install verify
```

*Note*: it's ok if the integration tests fail when running w/ mvnd. In fact, this is what I want to investigate (whether we can make them work in this scenario or not).

**Comment by khmarbaise** [comment]
Hi,

> Steps to reproduce:
> 
> ```
> git clone https://github.com/orpiske/camel-kafka-connector.git --branch mvnd-issue-86
> cd camel-kafka-connector
> mvnd -U -DskipIntegrationTests=false clean install verify

Is there a reason why you use `install verify` ? This will repeat several parts for the build life cycle...

I would suggest to use only `clean verify`...

**Comment by orpiske** [comment]
> Hi,
> 
> > Steps to reproduce:
> > ```
> > git clone https://github.com/orpiske/camel-kafka-connector.git --branch mvnd-issue-86
> > cd camel-kafka-connector
> > mvnd -U -DskipIntegrationTests=false clean install verify
> > ```
> 
> Is there a reason why you use `install verify` ? This will repeat several parts for the build life cycle...

Thanks for the explanation about repeating the cycles! I wasn't aware about it. Using the `install verify` is just an old habit ... 

> 
> I would suggest to use only `clean verify`...

So, I tried with just "clean install" but it didn't seem to have affected the behavior. The tests still did not execute. I am wondering if we have something on our pom that would cause it to skip under these circumstances.

**Comment by ppalaga** [comment]
Now I realized you pass `-DskipIntegrationTests=false`. Indeed it is not honored in 0.10.0. The root cause is described in https://github.com/mvndaemon/mvnd/issues/157#issuecomment-717102442 . #157 is fixed in the master. Could you plz try with a recent build from the CI - e.g. this one https://github.com/mvndaemon/mvnd/actions/runs/334727512

**Comment by orpiske** [comment]
> Now I realized you pass `-DskipIntegrationTests=false`. Indeed it is not honored in 0.10.0. The root cause is described in [#157 (comment)](https://github.com/mvndaemon/mvnd/issues/157#issuecomment-717102442) . #157 is fixed in the master. Could you plz try with a recent build from the CI - e.g. this one https://github.com/mvndaemon/mvnd/actions/runs/334727512

That was right on point. Using the master does make our tests run! Thanks Peter, I'll make sure to upgrade once the new version is out.

**Comment by ppalaga** [comment]
OK, let's close this as a duplicate of #157.
